### ISSUE TITLE
Not playing m4a sent from Android

### ISSUE THREAD

**Issue by dreamflasher** [signalapp/Signal-iOS#3590]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Audio files in m4a format from Android can not be played.

Creating this ticket as discussed here: https://github.com/signalapp/Signal-Android/issues/7889#issuecomment-397748415 (apparently it's not an Android bug, but on iOS side, with wrong mime types).

### Steps to reproduce
- send an audio file in m4a format from Android, eg.: http://techslides.com/demos/samples/sample.m4a

**Actual result:** Describe here what happens after you run the steps above (i.e. the buggy behaviour)
- the iOS client believes it is a mp3 file and can't play it

**Expected result:** Describe here what should happen after you run the steps above (i.e. what would be the correct behaviour)
- be able to play the m4a file

### Device info

**Device**: iPhone 8

**iOS version**: 11.3.1

**Signal version:** 2.26.0

**Comment by brokenalarms** [comment]
Hit this same issue, confirm it is still a bug 3 years later.

**Comment by baseljd** [comment]
Same issue happening with me when sending voice notes from the Signal CLI to iPhone devices. The iPhone device shows it as an .oga downloadable file.

**Comment by brokenalarms** [comment]
Yes, it has been occurring for years, is blocking me from using signal and
from the looks of it is stopping many other people as well. Seems like one
to prioritize?

**Comment by meg1502** [comment]
This is a big issue especially with the Google Recorder app saving files in m4a

**Comment by dreamflasher** [comment]
push

**Comment by brokenalarms** [comment]
still an issue. A major issue. is there an active dev we can ping here to get some awareness?

**Comment by robin13** [comment]
Can this please be re-opened? Still a very bothersome issue!

**Comment by EvanHahn-Signal** [comment]
We can't reproduce this, even with the .m4a provided above. Could someone provide a file that reproduces this issue, or steps we can use to replicate the issue?

**Comment by brokenalarms** [comment]
I got my friends to migrate from Signal back to Whatsapp due to this issue being present for so long and no fix. We were musicians collaborating with audio so this was kind of crucial. So I don't have it installed any more, sorry.

**Comment by danopia** [comment]
I recorded a second of room noise in Google Recorder to have a minimal file here: https://drive.google.com/file/d/1S8uiAdoj1Xc9WRK43wYLZsUmAx2DEFl3/view?usp=drivesdk

It sends like this on Pixel:
![Screenshot_20221118-172618.png](https://user-images.githubusercontent.com/40628/202753858-c1b19803-77fc-492a-98c7-50c41c23d276.png)

It shows up like this on iPhone:
![markup_1682.png](https://user-images.githubusercontent.com/40628/202754213-88ff3655-aede-4c92-a5fd-0cf4bb96c1b0.png)

I originally had the problem when sharing recorded audio from a different app (Barkio).

The receiving end cannot access the audio in any way and I've been working around by emailing iPhone users files instead of using Signal.

**Comment by robin13** [comment]
The issue still exists. There are now examples. Could the issue please be re-opened @EvanHahn-Signal ?

**Comment by sashaweiss-signal** [comment]
I wasn't able to reproduce either by using the steps above (sending an audio file taken using Google Recorder on a Pixel) or by sending the audio file linked in Google Drive or from techsamples.com. Do those steps still reproduce for others, sending and (more relevant) receiving using the latest versions of Signal?

Since this appears to continue to be an issue for some, I'll remove the `stale` label and reopen so we can hopefully get a repro.

**Comment by danopia** [comment]
I just repro'd with my chat partner after they updated to the latest Signal from the App Store. I simply downloaded and shared the previously attached file.

**Comment by robin13** [comment]
Just in case it is related to iPhone / iOS version, here's another data point (from a friend I sent audio to) where it is not working:
iPhone 12
iOS 16.3.1

**Comment by Anduril97** [comment]
Still an ongoing issue!!

**Comment by schobernoise** [comment]
Same here!!

**Comment by Miguel-Signal** [comment]
Hi @schobernoise @Anduril97 

Could you please email support@signal.org with a link to your debug logs and reference this Github issue?
To share a debug log from Signal iOS, tap your profile, Settings, Help, Submit Debug Log, Copy Link,

**Comment by schobernoise** [comment]
Sent two Debug Logs, one for Sender and one for Receiver.

**Comment by muc40** [comment]
Apparently this issue is still not resolved. Could you please confirm? Is there a workaround?

**Comment by Anduril97** [comment]
The issue is definitely not resolved for me

**Comment by schobernoise** [comment]
The issue occasionally persists, but far less frequent.

**Comment by spyro2000** [comment]
6 (SIX) years later, still the same problem. Seriously, WTF? ;D

**Comment by xantonin** [comment]
Just ran into the same issue. I (Android) sent my girlfriend an MP3 and she could play it in the app (so we know Android isn't messing with it). In the app it even sees the file as MP3, but when she saved it it saved the file as saved as MPG, and wouldn't play.

So I did another test using the Microphone button to record an audio directly in the app. She could play it but when she went to save the file it saved as M4A and didn't work.

In both cases renaming the file to MP3 on the iPhone worked and allowed the audio to play. 

(Oddly when I saved the Signal recorded audio on my Android it saved as OGA though)

**Comment by spyro2000** [comment]
Saving MP3 as. mpg sounds just wrong (is actually an MPEG-1 video file)

IHMO correct would be
- AAC - > *. aac (mimetype 'audio/aac') 
- MP3 - > *. mp3 (mimetype 'audio/mpeg')
